projects/plugins/jetpack/modules/stats/class-jetpack-stats-dashboard-widget.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Adds the Jetpack stats widget to the WordPress dashboard.
+ *
+ * @package jetpack
+ */
+
+use Automattic\Jetpack\Assets\Logo as Jetpack_Logo;
+use Automattic\Jetpack\Status;
+
+/**
+ * Class that adds a adds the Jetpack stats widget to the WordPress dashboard.
+ */
+class Jetpack_Stats_Dashboard_Widget {
+
+	/**
+	 * Indicates whether the class initialized or not.
+	 *
+	 * @var bool
+	 */
+	private static $initialized = false;
+
+	/**
+	 * Initialize the class by calling the setup static function.
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		if ( ! self::$initialized ) {
+			self::$initialized = true;
+			self::wp_dashboard_setup();
+		}
+	}
+
+	/**
+	 * Sets up the Jetpack Stats widget in the WordPress dashboard.
+	 */
+	public static function wp_dashboard_setup() {
+		if ( Jetpack::is_connection_ready() ) {
+			add_action( 'jetpack_dashboard_widget', array( __CLASS__, 'dashboard_widget_footer' ), 999 );
+		}
+
+		if ( has_action( 'jetpack_dashboard_widget' ) ) {
+			$jetpack_logo = new Jetpack_Logo();
+			$widget_title = sprintf(
+				// translators: Placeholder is a Jetpack logo.
+				__( 'Stats by %s', 'jetpack' ),
+				$jetpack_logo->get_jp_emblem( true )
+			);
+
+			// Wrap title in span so Logo can be properly styled.
+			$widget_title = sprintf(
+				'<span>%s</span>',
+				$widget_title
+			);
+
+			wp_add_dashboard_widget(
+				'jetpack_summary_widget',
+				$widget_title,
+				array( __CLASS__, 'dashboard_widget' )
+			);
+			wp_enqueue_style( 'jetpack-dashboard-widget', plugins_url( 'css/dashboard-widget.css', JETPACK__PLUGIN_FILE ), array(), JETPACK__VERSION );
+			wp_style_add_data( 'jetpack-dashboard-widget', 'rtl', 'replace' );
+
+			// If we're inactive and not in offline mode, sort our box to the top.
+			if ( ! Jetpack::is_connection_ready() && ! ( new Status() )->is_offline_mode() ) {
+				global $wp_meta_boxes;
+
+				$dashboard = $wp_meta_boxes['dashboard']['normal']['core'];
+				$ours      = array( 'jetpack_summary_widget' => $dashboard['jetpack_summary_widget'] );
+
+				$wp_meta_boxes['dashboard']['normal']['core'] = array_merge( $ours, $dashboard );
```

Comment:
PHPCS gives me:

> Overriding WordPress globals is prohibited. Found assignment to $wp_meta_boxes

Don't know if we can ignore this or need to find another solution.

Reply:
Looking, what about using `add_meta_box`.

`dashboard` would be `$screen->id`, `normal` would be `$context`, `core` would be `priority`, etc.

Looking in the commit history, https://github.com/Automattic/jetpack/pull/1716 ... doesn't really tell us much about what's going on.

Reply:
Doing some more testing, and I can't find a scenario where this is actually called. If we're not connected, the action isn't registered and the subsequent code block that checks for that action doesn't run. 

We could probably move it around, but I think we can just remove it. What do y'all think?

Reply:
This block was removed, see: https://github.com/Automattic/jetpack/pull/20676#discussion_r711384296